pkg/appgw/internaltypes.go

```diff
@@ -76,7 +89,7 @@ func generateHTTPSettingsName(serviceName string, servicePort string, backendPor
 }
 
 func generateProbeName(serviceName string, servicePort string, ingress string) string {
-	return fmt.Sprintf("%s-%v-%v-pb-%s", agPrefix, serviceName, servicePort, ingress)
+	return governor(fmt.Sprintf("%s-%v-%v-pb-%s", agPrefix, serviceName, servicePort, ingress))
```

Comment:
we should add a log mentioning that we shortened the name for this ingress/rule/path/backend(service)

Reply:
SGTM
Here is a sample log line: `•••••••••••I0514 12:27:22.761899   30440 internaltypes.go:66] Prop name xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx with length 81 is longer than 80; Transformed to xxxxxxxxxxxxxxx-4d1f7bb876c6aae013f06a7430b8c6545ff30d8f9252838eb18b6357c1a2ba13`

Reply:
Having a shorter hash like 128-bit and removing the ingress prefix, this would look better.

Reply:
I was actually thinking about using MD5 -- https://github.com/Azure/application-gateway-kubernetes-ingress/pull/128#pullrequestreview-236403946
Agreed - no need for a crytpographically strong hash - as long as we have a deterministic hash with some  fairly low probability of collision - I think MD5 here is perfect. 32 characters for the hash, 1 for the separator, 47 for the human readable part.